File: BackEnd/DotNet/WebDigital/WebDigital/dto/ProductListDto.cs
namespace WebDigital.dto
{
    public class ProductListDto
    {
        public int ProductId { get; set; }

        public string ProductName { get; set; } = null!;

        public string? ProductDescriptionShort { get; set; }

        public string? ProductImage { get; set; }

        public decimal ProductBaseprice { get; set; }

        public decimal? ProductOfferprice { get; set; }

        public decimal? DiscountPercent { get; set; }

        public bool IsRentable { get; set; }

        public bool IsLibrary { get; set; }

        public string? AuthorName { get; set; }

        public string? GenreName { get; set; }
    }
}

File: BackEnd/Java/product-service/src/main/java/com/digitalshelf/productservice/dto/ProductListDto.java
package com.digitalshelf.productservice.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

import com.fasterxml.jackson.annotation.JsonProperty;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProductListDto {
    private Integer productId;
    private String productName;
    private String productDescriptionShort;
    private String productImage;
    private BigDecimal productBaseprice;
    private BigDecimal productOfferprice;
    private BigDecimal discountPercent;
    private String authorName;
    private String genreName;
    
    @JsonProperty("isRentable")
    private boolean isRentable;

    @JsonProperty("isLibrary")
    private boolean isLibrary;
}